frontend/src/metabase/querying/filters/types.ts

```diff
@@ -104,4 +104,13 @@ export type MonthYearPickerValue = {
   month: number; // 1-12
 };
 
-export type DateFilterValue = DatePickerValue | MonthYearPickerValue;
+export type QuarterYearPickerValue = {
+  type: "quarter";
+  year: number;
+  quarter: number; // 1-4
```

Comment:
It'll be more useful as JSDoc (otherwise it's not shown in hints)

```suggestion
  /**
   * 1-4
   */
  quarter: number;
```